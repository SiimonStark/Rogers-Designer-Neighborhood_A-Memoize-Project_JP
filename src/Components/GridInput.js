import React, { Component } from 'react';

class GridInput extends Component {
  constructor() {
    super()
    this.state = {
      gridField0: {gridRule: '', gridValue: ''},
      inputCounter: 0,
      inputMapList: [0]
    }
  }
  handleGridChange = (e) => {
    const {value, id, className} = e.target
    let gridRule = this.state[id].gridRule
    let gridValue = this.state[id].gridValue;

    className === 'gridRule'
      ? gridRule = value
      : gridValue = value

    this.setState({[id]: {gridRule, gridValue}});
  }
  addInputField = () => {
    let counter = this.state.inputCounter + 1;
    let list = this.state.inputMapList;
    let fields = `gridField${counter}`;
    let newInput = {gridRule: '', gridValue: ''};

    list.push(counter)

    this.setState({inputCounter: counter, inputMapList: list, [fields]: newInput});
  }
  render() {
    console.log('GridState: ',this.state)

    return (
      <form className="GridInput">
        <h4>Grid-Input:</h4>
        {this.state.inputMapList.map((inputField, ind) => { return (
          <div className={'inputGroup' + ind}>
            <input id={'gridField' + ind} onChange={this.handleGridChange}  className="gridRule" type="text" placeholder="Rule" />
            <input id={'gridField' + ind} onChange={this.handleGridChange}  className="gridValue" type="text" placeholder="Value" />
          </div>
        )})}
        <input id="addInput" onClick={this.addInputField} value="+" type="button" />
      </form>
    );
  }
}

export default GridInput;