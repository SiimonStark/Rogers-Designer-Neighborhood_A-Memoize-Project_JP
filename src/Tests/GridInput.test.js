import React from "react";
import GridInput from "../Components/GridInput";
import { shallow } from "enzyme";

let updateInputsMock = jest.fn();
let mockState = {
  gridField0: { gridRule: "", gridValue: "" },
  inputCounter: 0,
  inputMapList: [0]
};

describe("Gridnput", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<GridInput updateInputs={updateInputsMock} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have default state", () => {
    expect(wrapper.state()).toEqual(mockState);
  });

  it.skip("when input is updated, it should invoke handleGridChange()", () => {
    const mockEvent = { target: { value: "d" } };
    wrapper.instance().handleGridChange(mockEvent);
  });
  
  it('should addInput field onClick of "+" button', () => {
    expect(wrapper.state('inputCounter')).toEqual(0)
    wrapper.find("#addInput").simulate("click");
    wrapper.find("#addInput").simulate("click");
    expect(wrapper.state("inputCounter")).toEqual(2);
    expect(wrapper.state("inputMapList").length).toEqual(3);
    expect(wrapper.find("div").length).toEqual(3);
    expect(wrapper).toMatchSnapshot();
  });
});
