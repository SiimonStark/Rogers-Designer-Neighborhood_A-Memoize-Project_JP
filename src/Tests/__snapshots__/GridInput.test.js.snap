// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gridnput should addInput field onClick of "+" button 1`] = `
<form
  className="GridInput"
>
  <h4>
    Grid-Input:
  </h4>
  <div
    className="gridInput inputGroup0"
    key="0"
  >
    <input
      className="gridRule"
      id="gridField0"
      onChange={[Function]}
      placeholder="Rule"
      type="text"
    />
    <span
      className="colon"
    >
      :
    </span>
    <input
      className="gridValue"
      id="gridField0"
      onChange={[Function]}
      placeholder="Value"
      type="text"
    />
  </div>
  <div
    className="gridInput inputGroup1"
    key="1"
  >
    <input
      className="gridRule"
      id="gridField1"
      onChange={[Function]}
      placeholder="Rule"
      type="text"
    />
    <span
      className="colon"
    >
      :
    </span>
    <input
      className="gridValue"
      id="gridField1"
      onChange={[Function]}
      placeholder="Value"
      type="text"
    />
  </div>
  <div
    className="gridInput inputGroup2"
    key="2"
  >
    <input
      className="gridRule"
      id="gridField2"
      onChange={[Function]}
      placeholder="Rule"
      type="text"
    />
    <span
      className="colon"
    >
      :
    </span>
    <input
      className="gridValue"
      id="gridField2"
      onChange={[Function]}
      placeholder="Value"
      type="text"
    />
  </div>
  <input
    id="addInput"
    onClick={[Function]}
    type="button"
    value="+"
  />
</form>
`;

exports[`Gridnput should match snapshot 1`] = `
<form
  className="GridInput"
>
  <h4>
    Grid-Input:
  </h4>
  <div
    className="gridInput inputGroup0"
    key="0"
  >
    <input
      className="gridRule"
      id="gridField0"
      onChange={[Function]}
      placeholder="Rule"
      type="text"
    />
    <span
      className="colon"
    >
      :
    </span>
    <input
      className="gridValue"
      id="gridField0"
      onChange={[Function]}
      placeholder="Value"
      type="text"
    />
  </div>
  <input
    id="addInput"
    onClick={[Function]}
    type="button"
    value="+"
  />
</form>
`;
